bind.go

```diff
@@ -39,6 +40,26 @@ func (bindRequest *SimpleBindRequest) encode() *ber.Packet {
 	return request
 }
 
+func (l *Conn) SimpleBindWithTimeout(simpleBindRequest *SimpleBindRequest, timeout time.Duration) (*SimpleBindResult, error) {
+	errChan := make(chan error, 1)
+	resChan := make(chan *SimpleBindResult, 1)
+	go func() {
+		res, err := l.SimpleBind(simpleBindRequest)
+		if err != nil {
+			errChan <- err
+		} else {
+			resChan <- res
+		}
+	}()
+	select {
+	case err := <-errChan:
+		return nil, err
+	case res := <-resChan:
+		return res, nil
+	case <-time.After(timeout):
+		return nil, NewError(ErrorNetwork, errors.New("ldap: timeout"))
```

Comment:
Does the connection become unusable in this case? What happens if the server responds after this timeout? What if another operation is in progress at that point?